src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat.pop());
+  const dateObj = {};
+
+  fromFormat.forEach(function (part, i) {
+    dateObj[part] = dateParts[i];
+  });
+
+  const convertYear = (year, format) => {
+    if (format === 'YYYY') {
+      return year.length === 2 ? (year < 30 ? '20' + year : '19' + year) : year;
+    }
+
+    return year.length === 4 ? year.slice(2) : year;
+  };
+
+  const resultParts = toFormat.slice(0, -1).map((part) => {
+    if (part === 'YYYY' || part === 'YY') {
+      const yearFormat = fromFormat.find((f) => f.includes('Y'));
```

Comment:
The use of `find` here assumes there will be only one format part containing 'Y', which is a fair assumption given the task description. However, if the formats were to become more complex in the future, this might not hold true. It's a good practice to code defensively and check for potential issues that could arise if the assumptions change.